test/functionality/r-bridge/executor.ts

```diff
@@ -0,0 +1,77 @@
+import {guard} from '../../../src/util/assert'
+import {assert} from 'chai'
+import semver from 'semver/preload'
+import {RShellExecutor} from '../../../src/r-bridge/shell-executor'
+import fs from 'fs'
+import {testRequiresNetworkConnection} from '../_helper/network'
+import {isInstallTest} from '../main.spec'
+import {parseCSV} from '../../../src'
+
+describe('RShellExecutor', function() {
+	it('R version', () => {
+		const version = new RShellExecutor().usedRVersion()
+		guard(version !== null, 'we should be able to retrieve the version of R')
+		assert.isNotNull(semver.valid(version), `the version ${JSON.stringify(version)} should be a valid semver`)
+		assert.isTrue(semver.gt(version, '0.0.0'), `the version ${JSON.stringify(version)} should not be 0.0.0`)
+	})
+
+	it('ignore errors', () => {
+		const executor = new RShellExecutor()
+
+		// check regular result when an error occurs
+		const error = executor.run('a', true)
+		assert.match(error, /Error.*'a'/g)
```

Comment:
this depends on the localization, so you require:

`options(warn=-1); invisible(Sys.setlocale("LC_MESSAGES", 'en_GB.UTF-8'))`

A test of the R sessions does exactly that (although i am still not 100% sure if the lines above are sufficient for every R version).